@danielfullmer On your branch (ed5bda15fa59b6cde68784ffe88ec83e67a64b19 to be precise) I still get the following build failure when running `nix-build -A pkgsCross.armv7l-hf-multiplatform.rpcsvc-proto`:

```
building
build flags: SHELL=/nix/store/k8p54jg8ipvnfz435mayf5bnqhw4qqap-bash-4.4-p23/bin/bash
make  all-recursive
make[1]: Entering directory '/build/source'
Making all in rpcgen
make[2]: Entering directory '/build/source/rpcgen'
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_clntout.o rpc_clntout.c
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_cout.o rpc_cout.c
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_hout.o rpc_hout.c
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_main.o rpc_main.c
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_parse.o rpc_parse.c
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_sample.o rpc_sample.c
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_scan.o rpc_scan.c
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_svcout.o rpc_svcout.c
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_tblout.o rpc_tblout.c
armv7l-unknown-linux-gnueabihf-gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c -o rpc_util.o rpc_util.c
armv7l-unknown-linux-gnueabihf-gcc  -g -O2   -o rpcgen rpc_clntout.o rpc_cout.o rpc_hout.o rpc_main.o rpc_parse.o rpc_sample.o rpc_scan.o rpc_svcout.o rpc_tblout.o rpc_util.o
make[2]: Leaving directory '/build/source/rpcgen'
Making all in rpcsvc
make[2]: Entering directory '/build/source/rpcsvc'
../rpcgen/rpcgen -h -o klm_prot.h klm_prot.x
/nix/store/k8p54jg8ipvnfz435mayf5bnqhw4qqap-bash-4.4-p23/bin/bash: ../rpcgen/rpcgen: cannot execute binary file: Exec format error
make[2]: *** [Makefile:582: klm_prot.h] Error 126
make[2]: Leaving directory '/build/source/rpcsvc'
make[1]: *** [Makefile:391: all-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:332: all] Error 2
error: --- Error --- build-remote
build of '/nix/store/7gznyxfrzy5z8c86d5kimyvzixc5dcn9-rpcsvc-proto-1.4.2-armv7l-unknown-linux-gnueabihf.drv' on 'ssh://builder-ng' failed: error: --- Error --- nix-store
builder for '/nix/store/7gznyxfrzy5z8c86d5kimyvzixc5dcn9-rpcsvc-proto-1.4.2-armv7l-unknown-linux-gnueabihf.drv' failed with exit code 2
error: --- Error ------------------------------------------------------------------------------------------------------------------- nix-build
error: --- Error --- nix-daemon
builder for '/nix/store/7gznyxfrzy5z8c86d5kimyvzixc5dcn9-rpcsvc-proto-1.4.2-armv7l-unknown-linux-gnueabihf.drv' failed with exit code 1
```

--- a/pkgs/tools/misc/rpcsvc-proto/default.nix
+++ b/pkgs/tools/misc/rpcsvc-proto/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, autoreconfHook }:
+{ stdenv, lib, fetchFromGitHub, autoreconfHook, buildPackages }:
 
 stdenv.mkDerivation rec {
   pname = "rpcsvc-proto";
@@ -13,7 +13,13 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "man" ];
 
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
   nativeBuildInputs = [ autoreconfHook ];
+
+  postPatch = lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
+    substituteInPlace rpcsvc/Makefile.am \
+      --replace '$(top_builddir)/rpcgen/rpcgen' '${buildPackages.rpcsvc-proto}/bin/rpcgen'
+  '';
 
   meta = with stdenv.lib; {
     homepage = "https://github.com/thkukuk/rpcsvc-proto";

